spec/02-integration/13-vaults/04-echo_spec.lua

```diff
@@ -0,0 +1,189 @@
+local helpers = require "spec.helpers"
+
+
+local ADMIN_HEADERS = {
+  ["Content-Type"] = "application/json",
+}
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("Vault configuration #" .. strategy, function()
+    local admin_client
+    local proxy_client
+
+    lazy_setup(function()
+      local bp = helpers.get_db_utils(strategy, { "vaults" }, { "secret-response" }, { "echo" })
+
+      local route = bp.routes:insert {
+        paths = { "/" },
+      }
+
+      bp.plugins:insert {
+        name    = "secret-response",
+        route   = { id = route.id },
+      }
+
+      assert(helpers.start_kong {
+        database = strategy,
+        prefix = helpers.test_conf.prefix,
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        plugins = "secret-response",
+        vaults = "echo",
+      })
+    end)
+
+    before_each(function()
+      admin_client = assert(helpers.admin_client())
+      proxy_client = assert(helpers.proxy_client())
+    end)
+
+    after_each(function()
+      if admin_client then
+        admin_client:close()
+      end
+      if proxy_client then
+        proxy_client:close()
+      end
+    end)
+
+    lazy_teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("is not sticky", function()
+      -- Create Vault:
+      local res = admin_client:put("/vaults/secrets", {
+        body = {
+          name = "echo",
+          config = {
+            prefix = "prefix",
+            suffix = "suffix",
+          },
+        },
+        headers = ADMIN_HEADERS,
+      })
+      assert.response(res).has.status(200)
+
+      -- Round 1:
```

Comment:
```suggestion
      -- Invoke a plugin that returns the secret (by reference) we passed as query arg.
```

Reply:
`Round 1` there means the requests made in Round 1, then change and same requests are made in round 2. Perhaps better to move those requests to a function.

Reply:
I moved the requests to separate function.